### DISCUSSION TITLE
Google Summer of Code 2023 & CNCF

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#848] [Google Summer of Code 2023]
Google Summer of Code has selected the CNCF as one of the mentoring organizations!

Between now and March 19, Potential GSoC contributors can discuss application ideas with the projects. You can find this list of ideas here: https://github.com/cncf/mentoring/blob/main/summerofcode/2023.md. Contributors, you can use this discussion to ask about and discuss these ideas, or you can reach out on the proposal's upstream issue (if you do use this thread, be sure to tag the mentor).

[GSoC Contributor/Student Guide](https://google.github.io/gsocguides/student/)
[Mentor Guide](https://google.github.io/gsocguides/mentor/)

## Important Upcoming Dates

For transparency here are all the dates that are coming up, for mentors, prospective contributors, and the CNCF org administrators.

**Now - March 19**: Mentors, proactive GSoC contributors will reach out to you to ask questions about [our proposals list](https://github.com/cncf/mentoring/blob/main/summerofcode/2023.md) and receive feedback from you so they can start crafting their project proposals.

**March 20 - April 4 18:00 UTC**: GSoC contributors will submit their proposals through the program website for mentors to give solid feedback on.

**April 4 - 26**: Mentorship WG team to review all submitted GSoC Contributor proposals and consider how many to want to select and how many we can handle (how many we have committed mentors for). Decide on the ranking of the GSoC Contributor slots. At least 1 mentor must be assigned to each project before it can be selected.

**April 27 18:00 UTC**: Deadline to submit ranked slot requests (CNCF org Admins enter these requests)

**April 27 - May 2**: Google Program Admins review and assign org slots

**May 3**: Organizations receive notification of their accepted GSoC 2023 Contributors

**May 4**: Accepted GSoC 2023 GSoC Contributor projects are announced

**May 4 - 28**: Community Bonding Period

**May 27**: Deadline to notify Google Admins of an inactive GSoC Contributor that you wish to remove from the program

**May 29**: Coding begins

**Comment by kannon92** [comment]
Thank you for posting this!  I was just about to post a discussion to get some clarity on this process.

**Comment by DataCrusade1999** [comment]
Is there any prerequisite that one should include in his proposal when applying to CNCF like a link to a merged PR?? I tried finding about it but couldn't.

**Reply by nate-double-u** [reply]
Hi @DataCrusade1999, I've updated the post to include links to [some GSoC guides](https://developers.google.com/open-source/gsoc/resources/guide). The student one has a great section on applying and has a section on proposal writing.

As for prerequisites, checkout the [GSoC FAQ entry for eligibility requirements](https://developers.google.com/open-source/gsoc/faq#what_are_the_eligibility_requirements_for_participation).

I also recommend reaching out to the mentors of the projects you're interested in and talking with them about what their expectations are.

**Comment by elijah0kello** [comment]
If I have questions about a certain idea, where do I post them? Here or on slack. If on slack, which channel? Thank you.

**Reply by aliok** [reply]
@elijah0kello Posting here is not a good idea IMO. I would check out the repositories for the projects and see how to reach out to the mentors and the community. Some communities prefer Slack, others prefer emails.

**Comment by Pratham-19** [comment]
Hi @nate-double-u , this is my first time contributing to a large open source organization. I find these projects interesting please tell me what can I do to get started and what can I do differently ? -

Keptn Plugin for Backstage
Vela IDE Plugins
Implement ClickHouse support
Add Kubectl Plugin for Armada
Build interfaces around Postgres for Armada
Thanks

**Reply by aliok** [reply]
I wrote a reply here for the same question: https://github.com/cncf/mentoring/discussions/848#discussioncomment-5118782

**Reply by Pratham-19** [reply]
thnx

**Comment by Chinwendu20** [comment]
Hello, I suggest that the organization's communication channel is included for each project on the project ideas page. I have seen one or two go into the cncf slack channel and they are confused on how to locate their organizations.

**Reply by nate-double-u** [reply]
We'll take that feedback into account for next year. In the meantime, each proposal has an upstream issue as well as mentor email addresses. I recommend following the link to the **upstream issues first** and seeing how each project prefers communication--the info may not be in the issue itself, but their repo, or their website, should have information on how best to engage their community.

**Comment by bwplotka** [comment]
Can we still add ideas to https://github.com/cncf/mentoring/blob/main/summerofcode/2023.md as mentioned in https://github.com/cncf/mentoring/discussions/777#discussion-4792008?

**Reply by nate-double-u** [reply]
Yes, please see my response here: https://github.com/cncf/mentoring/discussions/777#discussioncomment-5196153

**Comment by heysujal** [comment]
Is it okay if I am a beginner to this project entirely, don't have prior experience with the tech stack, and do not have any ideas to contribute?
Can I just work on the ideas given? For example, I want to contribute to buildpacks project.

**Reply by aliok** [reply]
@heysujal Thanks for your interest in GSoC at CNCF.

Items listed in GSoC Project Ideas document (https://github.com/cncf/mentoring/blob/main/summerofcode/2023.md#cloud-native-buildpacks) specify the recommended skills for each project idea. It will be the mentors who will do the first selection of contributors to the program, by evaluating their proposals to implement the project idea, their community interaction, their enthusiasm and similar attributes. (I said "first selection" as there are other criteria: CNCF's and Google's review and slot limitations.)

If you would like more information, Google has a [guide](https://google.github.io/gsocguides/student/am-i-good-enough) which might be answering your questions in detail.

Note: I just noticed the 2nd CloudPacks idea is missing the "recommended skills" information. I created a [ticket](https://github.com/cncf/mentoring/issues/874) about it.

**Reply by heysujal** [reply]
got it

**Comment by a-verma26** [comment]
Hi @nate-double-u , @aliok. 
I am reaching out to express my interest in contributing to your organization as a participant in the upcoming program. I have prior work experience in Typescript, Javascript, React,SQL, Java, and Kubernetes, and I believe I can make a valuable contribution to your team's efforts and to gain valuable knowledge and experience through the Google Summer of Code program.

I am looking forward to contribute to Falco and Keptn project.

**Reply by aliok** [reply]
hi @a-verma26 .

It is very nice to see that you're interested in GSoC at Keptn 👍 

I would recommend you to reach out to Keptn community. You may find the communication channels of Keptn community in their repository.

**Comment by Aman123lug** [comment]
@aliok i just want to ask where can I submit my early proposal for feedback my project is Knative. i didn't send any PR is this affect in my chance?

**Reply by kannon92** [reply]
I probably think you’d want feedback from Knative mentor. I’d suggest reaching out. I am a mentor on the Armada project so I’ve been getting google docs of proposals so I can give feedback.

**Reply by Aman123lug** [reply]
@kannon92 sir i want to contribute on this project but i didn't send my PR or something else is i need to complete the task first in order to selection for gsoc

**Comment by merhmood** [comment]
@nate-double-u  please, i couldn't find any template to use when submitting a contributor proposal

**Reply by nate-double-u** [reply]
We don't have a CNCF template for GSoC. We have been [discussing creating one](https://github.com/cncf/mentoring/issues/622), but it's not been completed, but there are some good resources there though.

**Comment by elijah0kello** [comment]
You can find a guide on how to write a proposal here.

https://google.github.io/gsocguides/student/writing-a-proposal

**Comment by nate-double-u** [comment]
Hi Everyone, a reminder for contributors: 

⚠️ Proposals are due **April 4 18:00 UTC**. Be sure to get those proposals in, the GSoC program is very strict with its due dates, so don't leave it to the last minute!

**Comment by merhmood** [comment]
has anyone heard from @jasondellaluce, was suppose to get feedback from him

**Reply by jasondellaluce** [reply]
Hey @merhmood, I answered you in private, sorry for the extra delay and thanks for addressing my previous round of feedback!

**Comment by kannon92** [comment]
Hey, I tried to login into the GSOC website for CNCF.  I notice that it only is showing 10 proposals now.  is there something wrong?

I counted them yesterday and I think there were over 90 proposals.

**Reply by nate-double-u** [reply]
Hi @kannon92, There is a bug with the ignore feature -- I've posted the GSoC admin's latest email over in https://github.com/cncf/mentoring/discussions/918 -- essentially clicking ignore then hides a proposal from everyone, not just the person clicking ignore.

We're going through to see if we can clean things up, and will be reaching out to GSoC  for help if we need. My understanding is that all the proposals are still there, they just aren't visible to everyone.

**Reply by kannon92** [reply]
I am able to find mine because I starred them.  But this may be a problem if someone didn't star them from a few days ago.

**Reply by aliok** [reply]
hi @kannon92 

We went through all proposals and removed them from the ignore list. They should be visible again.

**Reply by kannon92** [reply]
Great! It looks much better now.

**Comment by DavidPHirsch** [comment]
@nate-double-u @aliok I applied to be a mentor and selected a project proposal for OpenFeature, but I am not listed as an assigned mentor

**Reply by aliok** [reply]
Thanks @DavidPHirsch for letting us know. I now assigned you as a mentor to the proposal.
I don't know how this happened though. Do you remember, when did you press the "Want to Mentor" button? 

I wonder if it was after us assigning mentors, or if it was a simple mistake on our side. I don't want to have this situation next year again.

**Reply by DavidPHirsch** [reply]
@aliok honestly not sure when I had clicked it

**Comment by jasondellaluce** [comment]
@nate-double-u @aliok: Hi, maybe this info is already documented and I was not able to find out about it, but does the CNCF plan to apply to the GSoC Mentor Summit as an org? If so, what is the application process for mentors?

**Reply by nate-double-u** [reply]
Hi @jasondellaluce, I'm not sure if there is an org option per se. I've added this topic to our agenda for discussion at our next mentorship wg meeting: https://hackmd.io/@tag-cs-mentoring-wg/monthly-meeting-2023

**Reply by jasondellaluce** [reply]
Thanks! Any news from the meeting discussion?

**Reply by nate-double-u** [reply]
Hi @jasondellaluce, i've reached out in CNCF slack. Sorry for the delay.
